Back-end/src/com/smartplanner/OptimalityCalculator.java

```diff
@@ -0,0 +1,114 @@
+package com.smartplanner;
+
+import java.time.Duration;
+import java.time.LocalTime;
+import java.util.ArrayList;
+
+public class OptimalityCalculator {
+    private TimeDistanceManager timeDistanceManager;
+    private int maxCommutesPerDay;
+    private int minTimeSpentOnOptimizedAtOnceInMinutes;
+    private int numberOfDaysInCycle;
+    private OptimizedActivity optimizedActivity;
+
+    public OptimalityCalculator(TimeDistanceManager timeDistanceManager, int maxCommutesPerDay, int minTimeSpentOnOptimizedAtOnceInMinutes,
+                                int numberOfDaysInCycle, OptimizedActivity optimizedActivity) {
+        this.optimizedActivity = optimizedActivity;
+        this.timeDistanceManager = timeDistanceManager;
+        this.maxCommutesPerDay = maxCommutesPerDay;
+        this.minTimeSpentOnOptimizedAtOnceInMinutes = minTimeSpentOnOptimizedAtOnceInMinutes;
+        this.numberOfDaysInCycle = numberOfDaysInCycle;
+    }
+
+    public OptimalTimetable calculate(ArrayList<TimetableEntry> timetable) {
+        ArrayList<ArrayList<Boolean>> optimalDecisionPoints = new ArrayList<ArrayList<Boolean>>(numberOfDaysInCycle);
+        int timeSpentInWorkInCycle = 0;
+
+        for (int cycleDayNumber = 0; cycleDayNumber < numberOfDaysInCycle; ++cycleDayNumber) {
+            GoToOptimizedActivityDecider decider = new GoToOptimizedActivityDecider(timetable, cycleDayNumber);
+            int maxForCurrentDay = 0;
+            ArrayList<Boolean> bestDecisionPoints = null;
+
+            while (decider.isNext()) {
+                ArrayList<Boolean> currDecisionPoints = decider.getNext();
+                int currentVal = calculateRegardingProvidedDecisionPoints(timetable, currDecisionPoints, cycleDayNumber);
+
+                if (currentVal > maxForCurrentDay) {
+                    bestDecisionPoints = currDecisionPoints;
+                    maxForCurrentDay = currentVal;
+                }
+            }
+
+            timeSpentInWorkInCycle += maxForCurrentDay;
+            optimalDecisionPoints.add(bestDecisionPoints);
+        }
+
+        return new OptimalTimetable(timeSpentInWorkInCycle, timetable, optimalDecisionPoints);
+    }
+
+    private int getNumberOfOnes(ArrayList<Boolean> decisionPoints) {
+        int numberOfOnes = 0;
+        for (Boolean decision : decisionPoints)
+            if (decision == true)
+                ++numberOfOnes;
+
+        return numberOfOnes;
+    }
+
```

Comment:
Could you break the lines in this method where the single line width exceeds 120 characters?